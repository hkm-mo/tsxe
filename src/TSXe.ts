
export { Fragment as Fragment } from "./Fragment";
export { RefObject, FunctionComponent,
    ClassComponent, PropsWithChildren } from "./interfaces"
export { Component } from "./Component";
export { createElement, createRef, render } from "./utilities";

import { Component } from "./Component";

/**
 * @deprecated To align with React, please use `TSXe.Component` instead
 */
export const TSXeComponent = Component;

/**
 * @deprecated To align with React, please use `JSX.Element` instead
 */
 export type TSXeNode = JSX.Element;