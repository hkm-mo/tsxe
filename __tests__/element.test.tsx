import TSXe from "../src";

const helloWorld = "Hello World";

beforeEach(() => {
    document.body.innerHTML = "";
});

test("create element", () => {
    const content = ["Hello"," ", "World"];

    document.body.appendChild(<div>{content}</div>);

    expect(document.body.innerHTML).toMatchSnapshot();
});

test("nested array", () => {
    const children = [
        ["Hello "],
        [["World", [<span>!</span>]]]
    ];

    document.body.appendChild(<div>{children}</div>);

    expect(document.body.innerHTML).toMatchSnapshot();
});


test("create Ref object", () => {
    const elmId = "refObjDiv";
    const ref = TSXe.createRef<HTMLDivElement>();

    document.body.appendChild(<div id={elmId} ref={ref}></div>);

    const elm = document.getElementById(elmId);

    expect(elm).toBe(ref.current);
});


test("create Ref func", () => {
    const elmId = "refFuncDiv";
    const ref = (ref: HTMLDivElement) => {
        const elm = document.getElementById(elmId);
        expect(elm).toBe(ref);
    };

    document.body.appendChild(<div id={elmId} ref={ref}></div>);
});

test("fragment", () => {
    // <span/> should not count as <body/>'s child
    document.body.appendChild(
        <>
            <div></div>
            <div></div>
            <div><span></span></div>
        </>
    );

    expect(document.body.innerHTML).toMatchSnapshot();
});

test("styles string", () => {
    const ref = TSXe.createRef<HTMLDivElement>();
    document.body.appendChild(<div ref={ref} style="margin-left: 10px"></div>);

    expect(document.body.innerHTML).toMatchSnapshot();
});

test("styles object", () => {
    const ref = TSXe.createRef<HTMLDivElement>();
    document.body.appendChild(<div ref={ref} style={{marginLeft: "10px"}}></div>);

    expect(document.body.innerHTML).toMatchSnapshot();
});


test("data attributes", () => {
    const ref = TSXe.createRef<HTMLDivElement>();
    document.body.appendChild(<div ref={ref} data-test-info={helloWorld}></div>);

    expect(ref.current?.dataset.testInfo).toBe(helloWorld);
});

test("dataset object", () => {
    const ref = TSXe.createRef<HTMLDivElement>();
    document.body.appendChild(<div ref={ref} dataset={{testInfo: helloWorld}}></div>);

    expect(ref.current?.dataset.testInfo).toBe(helloWorld);
});