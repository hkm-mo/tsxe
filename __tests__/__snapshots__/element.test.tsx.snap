// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create element 1`] = `"<div>Hello World</div>"`;

exports[`fragment 1`] = `"<div></div><div></div><div><span></span></div>"`;

exports[`nested array 1`] = `"<div>Hello World<span>!</span></div>"`;

exports[`styles object 1`] = `"<div style=\\"margin-left: 10px;\\"></div>"`;

exports[`styles string 1`] = `"<div style=\\"margin-left: 10px\\"></div>"`;
